include/deal.II/fe/fe_system.h

```diff
@@ -970,6 +970,20 @@ class FESystem : public FiniteElement<dim,spacedim>
       base_elements;
 
   /**
+   * An index table that maps generalized support points of a base element
+   * to the vector of generalized support points of the FE System.
+   * It holds true that
+   * @code
+   *   auto n = generalized_support_points_index_table[i][j];
+   *   generalized_support_points[n] ==
+   *           base_elements[i].generalized_support_points[j];
+   * @endcode
+   * for each base element (indexed by i) and each g. s. point of the base
+   * element (index by j).
+   */
+  std::vector<std::vector<std::size_t>> generalized_support_points_index_table;
```

Comment:
@bangerth With this information at hand it should be straight forward to reimplement `convert_..._to_nodal_values` for the general case :-) (Will work on that tomorrow).

Reply:
Yes, good idea putting this into a table. This way, internally you can adopt any notation you want.

Have you thought about the fact that some base elements may have a multiplicity, and consequently you will have to decide whether the number of g.s.points for the `FESystem` element equals the sum over the number of such points for the base elements, or the number of such points for the base elements times their respective multiplicity? Right now, you go with the former route.

This allows to do some things more efficiently -- for example, you won't have to evaluate a function you want to interpolate onto a system element multiple times at the same location. On the other hand, it is not what we currently do with "regular" support points. Either way, it may be worth adding something along this discussion to the documentation here.

Reply:
The very idea is that the vector of general support points shall have no repeated point.
That will make a lot of things much easier. For example, we currently have the same logic of creating a representative set of support points in at least two different places for the interpolate function alone.

By basing the interpolation on generalized support points it is straightforward to eliminate this issue.

Reply:
Yes, I agree. I don't object, I just want us to remember to document this somewhere.

Reply:
This is now documented in the description of FiniteElement::get_generalized_support_points().